src/main/java/ro/jademy/contactlist/models/PhoneBook.java

```diff
@@ -20,6 +27,311 @@
+                //TODO
+                break;
+            case 9: // Return to Contact Menu
+                break;
+            default: // For invalid inputs
+                System.out.println("Invalid input. Please, choose between [1-9] only!");
+        }
+    }
+
+    private void contactMenu(byte option) {
+        switch (option) {
+            case 1: // Edit Contact
+                displayEditMenu();
+                byte input = INPUT.nextByte();
+                editContact(input);
+                break;
+            case 2: // Remove Contact
+                if (contacts.contains(searchForContact)) {
+                    contacts.remove(searchForContact);
+                    System.out.println("Contact deleted!");
+                    searchForContact = null;
+                    break;
+                }
+            case 3: // Add to Black List
+                //TODO
+                 /*
+                 creating a set that store the contact.
+                 If the Black list is empty, display the current contact added
+                 If not, display all contacts existing in the entire Black List
+                 */
+                break;
+            case 4: // Remove from Black List
+                //TODO
+                // deleting a contact from set
+                break;
+            case 5: // Add to Favorites
+                searchForContact.setGroup(Group.FAVORITE);
+                System.out.println("Contact added to Favorites!");
+                break;
+            case 6: // Removed from Favorites
+                searchForContact.setGroup(Group.MY_CONTACTS);
+                System.out.println("Contact removed from Favorites!");
+                break;
+            case 7: // Return to Main Menu
+                break;
+            default: // For invalid inputs
+                System.out.println("Invalid input. Please, choose between [1-7] only!");
+        }
+    }
 
+    private void addNewContact() {
+        Contact addNewContact = new Contact();
+        System.out.println("Country Code:");
+        String countryCode = INPUT.next();
+        System.out.println("Phone Number:");
+        String newPhoneNumber = INPUT.next();
+        addNewContact.setPhoneNumber(new PhoneNumber(countryCode, newPhoneNumber));
+        System.out.println("First Name:");
+        String newFirstName = INPUT.next();
+        addNewContact.setFirstName(newFirstName);
+        System.out.println("Last Name:");
+        String newLastName = INPUT.next();
+        addNewContact.setLastName(newLastName);
+        contacts.add(addNewContact);
+    }
 
+    private Set<Contact> getContactsByFirstLetter() {
+        System.out.println("Enter the first letter of the first name:");
+        String letter = INPUT.next();
+        Set<Contact> contactSubSet = contacts.stream()
+                .filter(contact -> contact.getFirstName().substring(0, 1).toLowerCase().equalsIgnoreCase(letter))
```

Comment:
You can remove .toLowerCase because you have .equalsIgnoreCase.

Reply:
> You can remove .toLowerCase because you have .equalsIgnoreCase.

You're right. Nice pointing it out. Cheers!

Reply:
You're right. Nice pointing it out. Cheers!